### ISSUE TITLE
[clang-repl] "Duplicate Definition of Symbol" for out of line definition for virtual function

### ISSUE THREAD

**Issue by Vipul-Cariappa** [llvm/llvm-project#141039]
clang-repl raises "Duplicate Definition of Symbol", when creating an object with an out-of-line virtual function.
```c++
clang-repl> struct Klass { virtual int get_value(); };
clang-repl> int Klass::get_value() { return 0; }
clang-repl> auto x1 = Klass();
error: Duplicate definition of symbol '_ZTV5Klass'
clang-repl> %quit
```
But if the virtual function in defined within the class, it does not raise the error:
```c++
clang-repl> struct Klass { virtual int get_value() { return 0; } };
clang-repl> auto x1 = Klass();
clang-repl> %quit
```
